libparsec/crates/testbed/src/template/events.rs

```diff
@@ -1875,14 +1894,25 @@ impl TestbedEventArchiveRealm {
  * TestbedEventNewShamirRecovery
  */
 
+pub struct TestbedEventNewShamirRecoveryCache {
+    pub certificates: Vec<TestbedTemplateEventCertificate>,
+    pub ciphered_data: Bytes,
+}
+
 #[derive(Serialize, Deserialize, Clone)]
```

Comment:
I get that the testbed is a mysterious place, but why does a testbed event needs to be serialized/ deserialized?

Reply:
Not sure I want an answer though

Reply:
Serialization is need for two things:
- to compute le CRC, this allows the rust code running the test and the testbed server to ensure they have the same template (otherwise a single change in the template definition would cause subtle de-synchronization that would make our test fail in all kind of weird ways)
- to transmit customization of a template from the rust code running the test to the testbed server (in practice the rust code serialize the additional events it has created in during the customization, then ask the testbed server to create a regular organization with a given template, then ask the testbed server to load the additional events)